src/main/java/swinglib/JPanelBuilder.java

```diff
@@ -139,58 +182,105 @@ public JPanelBuilder addWest(final Component child) {
     return this;
   }
 
+  public JPanelBuilder addWestIf(final boolean condition, final Supplier<Component> componentSupplier) {
+    if(condition) {
+      addWest(componentSupplier.get());
+    }
+    return this;
+  }
+
+  public JPanelBuilder addCenterIf(final boolean condition, final Supplier<Component> componentSupplier) {
+    if(condition) {
+      addCenter(componentSupplier.get());
+    }
+    return this;
+  }
+
+
   public JPanelBuilder addCenter(final Component child) {
     components.add(new Pair<>(child, BorderLayoutPosition.CENTER));
     return this;
   }
 
-
   public JPanelBuilder add(final Component component) {
     components.add(new Pair<>(component, BorderLayoutPosition.DEFAULT));
     return this;
   }
 
   /**
    * Specify a grid layout with a given number of rows and columns.
-   * 
+   *
    * @param rows First parameter for 'new GridLayout'
    * @param columns Second parameter for 'new GridLayout'
    */
   public JPanelBuilder gridLayout(final int rows, final int columns) {
-    layout = Layout.GRID;
-    this.gridRows = rows;
-    this.gridColumns = columns;
+    layout = new GridLayout(rows, columns);
     return this;
   }
 
-  public JPanelBuilder border(final BorderType borderType) {
-    this.borderType = borderType;
+  public JPanelBuilder borderEmpty(final int borderWidth) {
```

Comment:
Note that this rename is currently causing a compilation error in `JPanelBuilderTest`.